PySCF.py

```diff
@@ -80,6 +80,33 @@ def is_consistent(self,other):
             .format(key,self.__dict__[key],other.__dict__[key]))
         return False
     return True
+
+  #---------------------------------------------------
+  def update_input(self, fname):
+    # check if input file already updated
+    lines = open(fname, 'r').read().split()
+    for line in lines:
+      if "mc.__dict__.update(lib.chkfile.load"  in line:
+        return
+
+    # update pyscf input file for restart, load from chkfile 
+    chkfile=fname+".chkfile"
+    outlines=[]
+    with open(fname, 'r') as inf:
+      for line in inf:
+        if "print('E(HF) =',m.kernel(init_dm))" in line:
+          outlines += ["m.__dict__.update(lib.chkfile.load('%s', 'scf'))\n"%chkfile]
```

Comment:
I think it should be 
m. init_guess_by_chkfile()

Using the libs directly probably will not be general.